### ISSUE TITLE
Library prefixes should also support the `$` token 

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/linter#294]
Related to https://github.com/dart-lang/linter/issues/287

This should be valid to help generated code avoid simple collisions:

``` dart
// Assume this file is foo.g.dart.
import 'foo.dart' as $source;

class FooImpl implements $source.Foo { ... }
```
